app/src/main/java/com/github/se/eventradar/viewmodel/EventDetailsViewModel.kt

```diff
@@ -1,91 +1,68 @@
-package com.github.se.eventradar.viewmodel
+package com.github.se.eventradar.model.event
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.github.se.eventradar.model.Location
-import com.github.se.eventradar.model.event.EventCategory
-import com.github.se.eventradar.model.event.EventTicket
-import com.google.firebase.Firebase
-import com.google.firebase.firestore.FirebaseFirestore
-import com.google.firebase.firestore.firestore
-import java.time.Instant
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
 import java.time.LocalDateTime
-import java.time.ZoneId
+import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 
-class EventDetailsViewModel(
-    private val db: FirebaseFirestore = Firebase.firestore,
-    eventId: String
+@HiltViewModel
+class EventDetailsViewModel
+@Inject
+constructor(
+    private val eventRepository: IEventRepository,
 ) : ViewModel() {
 
-  private val _uiState = MutableStateFlow(initialEventState())
```

Comment:
These shouldn't be deleted...

Reply:
If the `toString(context: Context)` method in the EventCategory enum simplifies the usage of localized string resources, I recommend keeping it. However, to enhance readability and avoid confusion with the standard toString() method, consider renaming it to something more descriptive like `getLocalizedName(context: Context)`

Reply:
> `initialEventState()` which was used to initialized the UiState does not exist anymore (initialization is now done directly in the dataclass) maybe you're talking about
> 
> ```kotlin
>   private val _uiState = MutableStateFlow(EventUiState())
>   val uiState: StateFlow<EventUiState> = _uiState
> ```
> 
> at line 31&32

I didn't see this. For visibility and consistency with other view models, this should be the first two lines of the view model.